packages/core/state/selection/actions.ts

```diff
@@ -515,6 +515,60 @@ export function toggleFileFolderCollapse(fileFolder: FileFolder): ToggleFileFold
     };
 }
 
+/**
+ * COLLAPSE_ALL_FILE_FOLDERS
+ * Intention to collapse all file folders
+ */
+export const COLLAPSE_ALL_FILE_FOLDERS = makeConstant(
+    STATE_BRANCH_NAME,
+    "collapse-all-file-folders"
+);
+
+export interface CollapseAllFileFoldersAction {
+    type: string;
+}
+
+export function collapseAllFileFolders(): CollapseAllFileFoldersAction {
+    return {
+        type: COLLAPSE_ALL_FILE_FOLDERS,
+    };
+}
+
+/**
+ * EXPAND_ALL_FILE_FOLDERS
+ * Intention to toggle the given file folder's collapsed state
+ */
+export const EXPAND_ALL_FILE_FOLDERS = makeConstant(STATE_BRANCH_NAME, "expand-all-file-folders");
+
+export interface ExpandAllFileFoldersAction {
+    type: string;
+}
+
+export function expandAllFileFolders(): ExpandAllFileFoldersAction {
+    return {
+        type: EXPAND_ALL_FILE_FOLDERS,
+    };
+}
+
+/**
+ * SET_IS_LOADING_FULL_TREE
+ * Intention to set the loading state for all directories in the tree.
+ * Differentiated from isLoading in directory hierarchy node sub-states
+ */
+export const SET_IS_LOADING_FULL_TREE = makeConstant(STATE_BRANCH_NAME, "set-is-loading-full-tree");
+
+export interface SetIsLoadingFullTreeAction {
+    payload: boolean;
+    type: string;
+}
+
+export function setIsLoadingFullTree(isLoadingFullTree: boolean): SetIsLoadingFullTreeAction {
```

Comment:
I'm not sure I understand why it is necessary to know if the full tree is loading vs the node?

Reply:
removed in [c16c1ae](https://github.com/AllenInstitute/biofile-finder/pull/446/commits/c16c1ae522161412502360a675bab15dbacbf44f)